cli/connect/console.go

```diff
@@ -0,0 +1,352 @@
+	}
+}
+
+func getExecutor(console *Console) (prompt.Executor, error) {
+	var executeFunc func(console *Console, in string) string
+	switch {
+	case console.protocol == PlainTextProtocol:
+		executeFunc = plainTextExecute
+	case console.protocol == BinaryProtocol:
+		executeFunc = binaryExecute
+	default:
+		return nil, fmt.Errorf("Unknown protocol: %s", console.protocol)
+	}
+
+	executor := func(in string) {
+		console.input += in + " "
+
+		if !inputIsCompleted(console.input, console.luaState) {
+			console.livePrefixEnabled = true
+			return
+		}
+
+		handleSetOutput(console, console.input)
+
+		if console.historyFilePath != "" {
+			common.AppendStringToFile(console.historyFilePath, console.input+"\n")
+		}
+
+		data := executeFunc(console, console.input)
+
+		fmt.Printf("%s\n", data)
+
+		console.input = ""
+		console.livePrefixEnabled = false
+	}
+
+	return executor, nil
+}
+
+func inputIsCompleted(input string, luaState *lua.LState) bool {
+	var err error
+
+	_, err = luaState.LoadString(input)
+	if err == nil || !strings.Contains(err.Error(), "at EOF") {
+		// valid Lua code or a syntax error not due to
+		// an incomplete input
+		return true
+	}
+
+	if _, err = luaState.LoadString(fmt.Sprintf("return %s", input)); err == nil {
+		// certain obscure inputs like '(42\n)' yield the
+		// same error as incomplete statement
+		return true
+	}
+
+	return false
+}
+
+func getCompleter(console *Console) (prompt.Completer, error) {
+	switch {
+	case console.protocol == PlainTextProtocol:
+		return getPlainTextCompleter(console), nil
+	case console.protocol == BinaryProtocol:
+		return getBinaryCompleter(console), nil
+	}
+
+	return nil, fmt.Errorf("Unknown protocol: %s", console.protocol)
+}
+
+func handleSetOutput(console *Console, in string) {
+	inWords := strings.Fields(in)
+
+	if len(inWords) != 3 {
+		return
+	}
+
+	if inWords[0] != "\\set" || inWords[1] != "output" {
+		return
+	}
+
+	if ConsoleOutputMode(inWords[2]) == ConsoleYAMLOutput {
+		console.outputMode = ConsoleYAMLOutput
+		return
+	}
+
+	if ConsoleOutputMode(inWords[2]) == ConsoleLuaOutput {
+		console.outputMode = ConsoleLuaOutput
+		return
+	}
+}
+
+func getPromptOptions(console *Console) []prompt.Option {
+	options := []prompt.Option{
+		prompt.OptionTitle(console.title),
+		prompt.OptionPrefix(console.prefix),
+		prompt.OptionLivePrefix(console.livePrefixFunc),
+
+		prompt.OptionHistory(console.historyLines),
+
+		prompt.OptionSuggestionBGColor(prompt.DarkGray),
+		prompt.OptionPreviewSuggestionTextColor(prompt.DefaultColor),
+
+		prompt.OptionCompletionWordSeparator(tarantoolWordSeparators),
+
+		prompt.OptionAddASCIICodeBind(
+			prompt.ASCIICodeBind{ // move to one word left
+				ASCIICode: []byte{0x1b, 0x62},
+				Fn: func(buf *prompt.Buffer) {
+					d := buf.Document()
+					wordLen := len([]rune(d.GetWordBeforeCursorWithSpace()))
+					buf.CursorLeft(wordLen)
+				},
+			},
+			prompt.ASCIICodeBind{ // move to one word right
+				ASCIICode: []byte{0x1b, 0x66},
```

Comment:
Im not sure, but its not clear what these bytes are. Maybe add some comments?

Reply:
Moved to variables.